### DISCUSSION TITLE
"wait" command and connection pool

### DISCUSSION THREAD

**Discussion by phoenix2x** [redis/go-redis#1989] [Q&A]
Question about the correct usage of "wait". According to the redis docs "[wait](https://redis.io/commands/wait)" command "returns when all the previous write commands sent in the context of **the current connection** are guaranteed to be received by the number of replicas". Since the library uses connection pooling how one can make sure that "wait" returns the number of replicas from a specific command and not something else that was running concurrently? 
In my understanding, both commands in the following code just pick available connections from the pool without any 
'stickiness' which might yield incorrect resulsts.
```
client.Set(ctx, "key", "val", time.Hour)
client.Wait(ctx, 5, 5*time.Second)
```

**Comment by ali-hammadi-ipsos** [comment]
I have found that you can use transaction in redis with this client https://github.com/redis/go-redis/blob/master/tx.go
